Add analytics_sample_rate to all integrations

delner opened the pull request:
Using the feature defined in #665, and as a follow up to #666, this pull request adds the event sample rate tag, if configured, to the following integrations:

- [x] active_model_serializers
- [x] active_record
- [x] aws
- [x] dalli
- [x] elasticsearch
- [x] excon
- [x] faraday
- [x] grape
- [x] graphql (https://github.com/rmosolgo/graphql-ruby/pull/2154)
- [x] grpc
- [x] http
- [x] mongodb
- [x] mysql2
- [x] rails
- [x] redis
- [x] rest_client
- [x] sequel
- [x] sinatra

brettlangdon commented:
We want to use `analytics_sample_rate` instead of `event_sample_rate`.

There should be 1 global configuration option `analytics` when set to `true` will enable all analytics for all web frameworks only (rack, rails, sinatra, etc)

Each integration will then have it's own 2 configuration options:
- `analytics` - boolean, whether to explicitly enable or disable for the integration (default: `nil` so we can test if it was explicitly set or not)
- `analytics_sample_rate` - float, the sample rate to use for that integration (default: `1.0`)

For web frameworks, we should set the tag with `analytics_sample_rate` if the `(global[:analytics] == true && integration[:analytics] != false) or integration[:analytics] == true`. Global is `true` and integration is `nil` or `true` or the integration is `true`.

For everything else, we should set the tag with `analytics_sample_rate` if `integration[:analytics] == true`

delner commented:
@brettlangdon All of the existing commits have been updated to match the name changes and logic, with the exception of the global flag (which does not yet exist.) I will carry on updating the remaining integrations from here.

delner commented:
@brettlangdon Ready for review. Worth pointing out Rails, Sinatra and all the other integrations in this PR are off by default, and will ignore the global flag unless explicitly enabled. Let me know if you'd like to change that behavior.

brettlangdon commented on lib/ddtrace/contrib/sinatra/tracer.rb:
```diff
@@ -41,6 +43,12 @@ def render(engine, data, *)
                   span.set_tag(Ext::TAG_TEMPLATE_NAME, data) if data.is_a? Symbol
+
+                  # Set analytics sample rate
+                  if Contrib::Analytics.enabled?(Datadog.configuration[:sinatra][:analytics_enabled])
```
We probably don't need this one, we likely only need the `Ext::SPAN_REQUEST` span to have the tag.

delner replied:
Removed this.

delner commented:
@brettlangdon Removed tags from render & cache spans, fixed a constant name, and changed defaults such that only Grape, GraphQL, Rack, Rails, & Sinatra will be toggled by the global flag.
